menu/cbs/menu_cbs_ok.c

```diff
@@ -3507,28 +3507,40 @@ void cb_generic_download(retro_task_t *task,
          dir_path = buf;
          break;
       }
+      case MENU_ENUM_LABEL_CB_SINGLE_THUMBNAIL:
+      {
+         dir_path = transf->path;
+         break;
+      }
       default:
          RARCH_WARN("Unknown transfer type '%s' bailing out.\n",
                msg_hash_to_str(transf->enum_idx));
          break;
    }
 
-   if (!string_is_empty(dir_path))
+   if (!string_is_empty(dir_path) && transf->enum_idx != MENU_ENUM_LABEL_CB_SINGLE_THUMBNAIL)
       fill_pathname_join(output_path, dir_path,
             transf->path, sizeof(output_path));
+   else if (transf->enum_idx == MENU_ENUM_LABEL_CB_SINGLE_THUMBNAIL)
+      strlcpy(output_path, transf->path, sizeof(output_path));
 
    /* Make sure the directory exists */
    path_basedir_wrapper(output_path);
 
+   RARCH_LOG("output path: %s\n", output_path);
```

Comment:
extra debug print

Reply:
Ah yeah.. had to add it because this whole function is retarded so I had to debug it.
Basically it sets output path, then it mutates it, and then it sets output path again 🙄